app/helpers/i18n_helper.rb

```diff
@@ -32,4 +34,38 @@ def language_options_for_select
     localised.sort_by!(&:first)
     options_for_select(localised, I18n.locale.to_s)
   end
+
+  def date_eras_gregorian(date)
+    return date unless date.is_a?(String)
+
+    # Match on digits+ only, strip starting '0's and if date' length 1-3 then build "date' + CE"
+    m = date.match(/^\d+$/)
+    return build_date(m[0], '+') if m && strip_leading_zeroes(m[0]).length < 4
+
+    # Match on '+' or '-' followed by date where date starts with digit followed by one or more digits or hyphens
+    m = date.match(%r{^\s*(\+|\-)\s*(\d[\d\-\/]*)\s*$})
+    return build_date(m[2], m[1]) if m
+
+    # Match abnormal dates, e.g. with format 'c. AD 46' etc.
+    ['a. de C.', 'c. AD'].each do |abbrev|
+      m = date.match(%r{^\s*#{abbrev}\s*(\d[\d\-\/])\s*$})
+      return build_date(m[1], '+') if m
+    end
+
+    # Otherwise do nothing.
+    date.strip
+  end
+
+  private
+
+  def build_date(year, prefix)
+    year = strip_leading_zeroes(year)
+    key = 'global.date.eras.gregorian.' + (prefix == '+' ? 'current' : 'before')
+    default = year + ' ' + (prefix == '+' ? 'CE' : 'BCE')
```

Comment:
This should not be necessary given that the English will always be present in localisations.